File: DevBhakti-master/devbhakti-frontend/check_json_encoding.py

import sys

file_path = r"c:\Users\admin\Downloads\DevBhakti-master\DevBhakti-master\devbhakti-frontend\src\locales\hi.json"

try:
    with open(file_path, 'rb') as f:
        data = f.read()
    
    index = 44281
    start = max(0, index - 50)
    end = min(len(data), index + 50)
    
    chunk = data[start:end]
    print(f"Byte chunk around {index}:")
    print(chunk)
    print("\nHex representation:")
    print(chunk.hex(' '))
    
    # Try to decode to see where it fails
    try:
        chunk.decode('utf-8')
        print("\nDecoding chunk directly worked (unexpectedly if it's invalid).")
    except UnicodeDecodeError as e:
        print(f"\nDecoding failed: {e}")

except Exception as e:
    print(f"Error: {e}")

File: DevBhakti-master/devbhakti-frontend/check_json_encoding.js

const fs = require('fs');
const filePath = 'c:\\Users\\admin\\Downloads\\DevBhakti-master\\DevBhakti-master\\devbhakti-frontend\\src\\locales\\hi.json';

try {
    const buffer = fs.readFileSync(filePath);
    const index = 44281;
    const start = Math.max(0, index - 100);
    const end = Math.min(buffer.length, index + 100);
    
    const slice = buffer.slice(start, end);
    console.log(`Byte chunk around ${index}:`);
    process.stdout.write(slice);
    console.log('\n\nHex representation:');
    console.log(slice.toString('hex').match(/.{1,2}/g).join(' '));
    
    try {
        const decoded = slice.toString('utf8');
        console.log('\nDecoded slice:', decoded);
    } catch (e) {
        console.log('\nDecoding failed:', e.message);
    }

    // Find the line number
    let lineCount = 1;
    for (let i = 0; i < index; i++) {
        if (buffer[i] === 10) lineCount++;
    }
    console.log(`\nEstimated line number for index ${index}: ${lineCount}`);

} catch (err) {
    console.error('Error:', err);
}
